DOCS-1524: Add PID config to motors and bases (#2547)
viamrobotics/docs: docs/components/motor/28byj48.md

          "in3": "<pin-number>",
          "in4": "<pin-number>"
        },

>         "ticks_per_rotation": <int>,

Reviewer: remove

>         "ticks_per_rotation": <int>,
>         "control_parameters": {
>             "p": <int>,
>             "i": <int>,
>             "d": <int>
>         }

Author: ```suggestion
        "ticks_per_rotation": <int>
```

          "in3": "13",
          "in4": "15"
        },

>         "ticks_per_rotation": 4096,
>         "control_parameters": {
>           "p": 1,
>           "i": 2,
>           "d": 3
>         }

Author: ```suggestion
        "ticks_per_rotation": 4096
```

| Name | Type | Inclusion | Description |
| ---- | ---- | --------- | ----------- |
| `board` | string | **Required** | `name` of the [board](/components/board/) the motor driver is wired to. |
| `pins` | object | **Required** | A JSON object containing the {{< glossary_tooltip term_id="pin-number" text="pin numbers" >}} the `in1`, `in2`, `in3`, and `in4` pins of the motor driver are wired to on the [board](/components/board/). |
| `ticks_per_rotation` | int | **Required** | Number of full steps in a rotation. The motor takes 5.625*(1/64)° per step. One full rotation (360°) is 4096 steps. |

> | `control_parameters` | object | Optional | A JSON object containing the coefficients for the proportional, integrative, and derivative terms. If you use `{ "p": 0, "i": 0, "d": 0 }`, `viam-server` auto-tunes the values and logs the values. To avoid retuning the control parameters constantly, copy the values from the log and add them to the configuration once tuned. For more information see [Control motor velocity with a sensor](#control-motor-velocity-with-a-sensor). |

Author: ```suggestion
| `control_parameters` | object | Optional | A JSON object containing the coefficients for the proportional, integrative, and derivative terms. If you use `{ "p": 0, "i": 0, "d": 0 }`, `viam-server` auto-tunes the values and logs the values. Tuning takes several seconds and spins the motors. To avoid tuning every time the robot starts up, copy the values from the log and add them to the configuration once tuned. For more information see [Control motor velocity with a sensor](#control-motor-velocity-with-a-sensor). |
```

> | `control_parameters` | object | Optional | A JSON object containing the coefficients for the proportional, integrative, and derivative terms. If you use `{ "p": 0, "i": 0, "d": 0 }`, `viam-server` auto-tunes the values and logs the values. Tuning takes several seconds and spins the motors. To avoid tuning every time the robot starts up, copy the values from the log and add them to the configuration once tuned. For more information see [Control motor velocity with a sensor](#control-motor-velocity-with-a-sensor). |

Author: ```suggestion
| `control_parameters` | object | Optional | A JSON object containing the coefficients for the proportional, integral, and derivative terms for linear and angular velocity. If you want these values to be auto-tuned, you can set all values to 0: `[ { "type": "linear_velocity", "p": 0, "i": 0, "d": 0 }, { "type": "angular_velocity", "p": 0, "i": 0, "d": 0 } ]`, and `viam-server` will auto-tune and logs the calculated values. Tuning takes several seconds and spins the motors. To avoid tuning every time the robot starts up, copy the values from the log and add them to the configuration once tuned. For more information see [Control motor velocity with a sensor](#control-motor-velocity-with-a-sensor). |
```

> | `control_parameters` | object | Optional | A JSON object containing the coefficients for the proportional, integral, and derivative terms. If you want these values to be auto-tuned, you can set all values to 0: `{ "p": 0, "i": 0, "d": 0 }`, and `viam-server` will auto-tune and logs the calculated values. Tuning takes several seconds and spins the motors. To avoid tuning every time the robot starts up, copy the values from the log and add them to the configuration once tuned. For more information see [Control motor velocity with a sensor](#control-motor-velocity-with-a-sensor). |

Reviewer: remove

Refer to your motor and motor driver data sheets for specifics.

![An example wiring diagram for a four wire 28BYJ-48 stepper motor driven by a ULN2003A driver chip breakout board. The driver is connected to a Raspberry Pi with four wires labeled IN1, IN2, IN3, and IN4. These are connected to Pi pins 11, 12, 13 and 15, respectively. A separate 12V power supply is attached to the motor driver to power the motor.](/components/motor/28byj48-wiring.png)

## Test the motor

{{< readfile "/static/include/components/test-control/motor-control.md" >}}

> 
> ## Control motor velocity with a sensor
> 
> {{< readfile "/static/include/components/motor-sensor.md" >}}

Author: ```suggestion
```
